### PR TITLE
Added profitability calcs and added profitThreshold in config

### PR DESCRIPTION
**Pull request by krasi-georgiev**
Each transaction cost is saved in the db and at the next transaction for
the same slot if the profit is below profitThreshold the transaction is
not executed.

Profit calculations are the same as in the tellor contract.

also moved the GWEI and WEI constants in the common package so that any
package can use them without circular dependencies.

closes: https://github.com/tellor-io/TellorMiner/issues/186

### CODE REVIEW

File: pkg/pow/cpuMiner.go
Comment (krasi-georgiev) [inline pkg/pow/cpuMiner.go:56]: moved to contractRPC

File: pkg/pow/pool.go
Comment (krasi-georgiev) [inline pkg/pow/pool.go:165]: not sure what this method actually does since it doesn' submit anything to the tellor contract.

File: pkg/pow/solutionHandler.go
Comment (krasi-georgiev) [inline pkg/pow/solutionHandler.go:55]: need to return the transaction data to use the gas cost when saving the tx cost in the db.

File: pkg/rpc/client.go
Comment (krasi-georgiev) [inline pkg/rpc/client.go:35]: This is added so that the eth client monitors the completion of a transaction. The transaction gas used value is then saved in the db.

File: pkg/rpc/contractRPC.go
Comment (krasi-georgiev) [inline pkg/rpc/contractRPC.go:28]: moved to the common package so that other packages can use it. Without moving it caused circular dependency.

File: pkg/rpc/contractRPC.go
Comment (krasi-georgiev) [inline pkg/rpc/contractRPC.go:50]: renamed it becuase it doesn't just prepare but also submits the tx.

File: pkg/tracker/ampl.go
Comment (krasi-georgiev) [inline pkg/tracker/ampl.go:33]: granularity is exposed as it is used when calculating the TRB price.

File: pkg/tracker/psrs.go
Comment (krasi-georgiev) [inline pkg/tracker/psrs.go:18]: this is exposed as a const so that can later get the granularity for the TRB to calculate the TRB price.

**Comment by krasi-georgiev** [comment]
There seems to be an issue with the submission as some transactions fail with `Fail with error 'Request ID is wrong'`  so please hold it for now.

https://rinkeby.etherscan.io/tx/0xc73bb449c9a4968a8fd4d4e379981db31dcf58645b3b13612eaf4fa74e23a34d

File: pkg/ops/miningManager.go
Comment (themandalore) [inline pkg/ops/miningManager.go:192]: move profit check here (below setting mgr.solution so it doesn't have to resolve the solution)

File: pkg/ops/miningManager.go
Comment (themandalore) [inline pkg/ops/miningManager.go:293]: add issue for event listener for nonceSubmitted events to determine slot Progress and whether it was yours.

File: pkg/config/config.go
Comment (themandalore) [inline pkg/config/config.go:88]: add example

File: pkg/ops/miningManager.go
Comment (krasi-georgiev) [reply pkg/ops/miningManager.go:293]: updated the issue on the tellor core repo as this needs to be added there.
https://github.com/tellor-io/TellorCore/issues/101

File: pkg/ops/miningManager.go
Comment (krasi-georgiev) [reply pkg/ops/miningManager.go:192]: I updated the logic please double check now. 
With the new logic it just blocks until a transaction is profitable and only then it submits.

One problem I see with this approach is that by the time a transaction is profitable the challenge might have changed and the solution would be invalid, but fixing this would require some greater refactorings and I would say to leave it for another PR. I also added a new issue to simplify the logic in the manager here as I think it can be much more simple than now.
tracking issue https://github.com/tellor-io/TellorMiner/issues/232

File: pkg/config/config.go
Comment (krasi-georgiev) [reply pkg/config/config.go:88]: added.

File: pkg/ops/miningManager.go
Comment (themandalore) [inline pkg/ops/miningManager.go:308]: will this exit if a new challenge get's passed down?

**Comment by krasi-georgiev** [comment]
> There seems to be an issue with the submission as some transactions fail with Fail with error 'Request ID is wrong' so please hold it for now.
rinkeby.etherscan.io/tx/0xc73bb449c9a4968a8fd4d4e379981db31dcf58645b3b13612eaf4fa74e23a34d

Further checking on this and after talking to @themandalore this seems normal when another miner has beet you to submitting a chalange.

File: pkg/ops/miningManager.go
Comment (krasi-georgiev) [reply pkg/ops/miningManager.go:308]: This will block until a SubmitSolution is profitable, 
Checks the current slot, checks the reward and if profitable it will unblock otherwise it waits 1 sec and checks again. Once it unblocks if the solution it submitted is already submitted I assume the transaction will fail and then it will need to calculate the new solution.

Comment (themandalore) [reply pkg/ops/miningManager.go:308]: Yeah, this will need to change. You want it to unblock if a new challenge comes down

Comment (krasi-georgiev) [reply pkg/ops/miningManager.go]: Completely refactored the code flow and it is not blocked and I think it is much easier to follow.

Please review only the mining manager changes. The changes regarding the logging are from a different PR and will disappear once that other PR is merged.

File: pkg/ops/miningManager.go
Comment (krasi-georgiev) [reply pkg/ops/miningManager.go]: Completely refactored the code flow and it is not blocked and I think it is much easier to follow.

File: pkg/ops/miningManager.go
Comment (themandalore) [inline pkg/ops/miningManager.go:142]: when would solution be nil here?

Comment (krasi-georgiev) [reply pkg/ops/miningManager.go:142]: `mgr.tasker.GetWork` sends nil to indicate that the challenge hasn't changed this signals the manager that can use the pendingSolution.

Here:
https://github.com/tellor-io/TellorMiner/pull/228/files?w=1#diff-455c0953444badebf81c8379c2da7d4ad3ba73c891e49ab10a4e13bd2a2a68b5R209

**Review by themandalore — APPROVED** [review]

**Pull request merged**
